src/utils/api.ts

```diff
@@ -1,2 +1,23 @@
 export const apiUrl = 'http://api.quran.com/api/v3';
-export const makeUrl = (path: string) => `${apiUrl}${path}`;
+
+/**
+ * Generates a url to make an api call to our backend
+ * @param path the path for the call
+ * @param parameters optional query params, {a: 1, b: 2} is parsed to "?a=1&b=2"
+ */
+export const makeUrl = (path: string, parameters?: Record<string, unknown>) => {
+  if (!parameters) {
+    return `${apiUrl}${path}`;
+  }
+
+  // The following section parses the query params for convenience
+  // E.g. parses {a: 1, b: 2} to "?a=1&b=2"
+  const parametersStringified = JSON.stringify(parameters);
```

Comment:
feels lame but i rather use a lib here like https://www.npmjs.com/package/qs or https://www.npmjs.com/package/query-string. They handle objects, arrays, etc etc

Reply:
qs is `10kb`, and query-string is `32kb`. I added qs since we'll likely reuse it in the future. Thank you for the suggestion!